langchain/src/chains/sql_database/sql_database_chain.ts

```diff
@@ -0,0 +1,115 @@
+import { DEFAULT_SQL_DATABASE_PROMPT } from './sql_database_prompt.js';
+import { BaseChain, ChainValues } from '../base.js';
+import { BaseMemory } from '../../memory/index.js';
+import { BaseLLM } from '../../llms/index.js';
+import { LLMChain } from '../llm_chain.js';
+import { SQLDatabase } from '../../sql_database.js';
+
+export type SerializedSqlDatabaseChain = {
+    _type: "sql_database_chain";
+    sql_database_chain_path?: string;
+};
+
+export class SqlDatabaseChain extends BaseChain {
+    llm: BaseLLM;
+
+   // LLM wrapper to use.
+    database: SQLDatabase;
+
+   // SQL Database to connect to.
+    prompt = DEFAULT_SQL_DATABASE_PROMPT;
+
+   // Prompt to use to translate natural language to SQL.
```

Comment:
I think the comments are all off by one?

Reply:
well spotted